### PR TITLE
[Backend] The DELETE teamMemebers API should also delete the respective file stored in the disk Fixed

### LINKED ISSUE #916: [Backend] The DELETE teamMemebers API should also delete the respective file stored in the disk
## Description
The [deleteTeamMembersAPI](https://github.com/HITK-TECH-Community/Community-Website/blob/main/backend/app/routes/teamMember/deleteTeamMember.js#L9) currently only deletes the data from db and so the image file stored in disk will keep on getting bigger.

## Expected Fix
The delete call should also delete the image file before deleting the data form the DB.

### PR DESCRIPTION
**Pull request by Hemu21**
## Issue that this pull request solves
[Issue Link](https://github.com/HITK-TECH-Community/Community-Website/issues/916) resolve #916 

 Closes: #916 

### Brief description of what is fixed or changed
The delete call should also delete the image file before deleting the data form the DB. added in the delete function.

## Types of changes

_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (Documentation content changed)
- [ ] Other (please describe): 

## Checklist

_Put an `x` in the boxes that apply_

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] My changes does not break the current system and it passes all the current test cases.

## Screenshots or video

https://github.com/HITK-TECH-Community/Community-Website/assets/106808387/003550b3-b761-4eb7-b5e5-98ce58c0fcbe

### CODE REVIEW

**Comment by Hemu21** [comment]
@Kajol-Kumari  can you please check and merge it.

File: backend/app/routes/teamMember/deleteTeamMember.js [hunk 1]
```diff
@@ -1,19 +1,33 @@
...
+    const payload = res.locals.decode;
+    const memberId = req.body.memberId;
+
+    if (payload.isSuperAdmin === false) {
+      return res.status(401).json({ error: 'You are not an admin' });
     }
-    catch(error) {
-        return res.status(500).json({error:"Some internal server error"});
+
+    const member = await teamMemberModel.findById(memberId);
```
Comment (Kajol-Kumari) [inline backend/app/routes/teamMember/deleteTeamMember.js:14]: Why so we need to make an extra db call here? can we use the `result`?

File: backend/app/routes/teamMember/deleteTeamMember.js [hunk 2]
```diff
@@ -1,19 +1,33 @@
...
+    if (!member) {
+      return res.status(401).json({ error: 'Invalid id' });
     }
-    
-}
+
+    const result = await teamMemberModel.findByIdAndDelete(memberId);
```
Comment (Kajol-Kumari) [inline backend/app/routes/teamMember/deleteTeamMember.js:19]: This result should have the detail of delete object, can't we do like below? - 
```
const result = await teamMemberModel.findByIdAndDelete(memberId);
if (!result) {
  return res.status(401).json({ error: 'Invalid id' });
}

const fileName = path.basename(result.image);
const imagePath = path.join(__dirname, '../../../uploads/teamMembersProfile/', fileName);
fs.unlink(imagePath, (err) => {
  if (err) {
    console.error('Failed to delete image file:', err);
  }
  console.log('Image file deleted:', imagePath);
});

return res.status(200).json({ message: 'Team member deleted successfully' });
```

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
@Hemu21 can you please address the requested changes?

**Comment by Hemu21** [comment]
@Kajol-Kumari  updated can you please check it.

File: backend/app/routes/teamMember/deleteTeamMember.js (hunk 2 above)
Comment (Hemu21) [reply backend/app/routes/teamMember/deleteTeamMember.js:12]: @Kajol-Kumari thank you. I forget that delete has the deleted object. so I make extra. Now i updated it.

**Review by Kajol-Kumari — APPROVED** [review]
LGTM 👍

**Pull request merged**